api/src/services/graphql/index.ts

```diff
@@ -196,1579 +102,7 @@ export class GraphQLService {
-			const aggregateKeys = Object.keys(query.aggregate ?? {});
-
-			result['map']((field: Item) => {
-				field['group'] = omit(field, aggregateKeys);
-			});
-		}
-
-		return result;
-	}
-
-	async resolveMutation(
-		args: Record<string, any>,
-		info: GraphQLResolveInfo,
-	): Promise<Partial<Item> | boolean | undefined> {
-		const action = info.fieldName.split('_')[0] as 'create' | 'update' | 'delete';
-		let collection = info.fieldName.substring(action.length + 1);
-		if (this.scope === 'system') collection = `directus_${collection}`;
-
-		const selections = this.replaceFragmentsInSelections(info.fieldNodes[0]?.selectionSet?.selections, info.fragments);
-		const query = this.getQuery(args, selections || [], info.variableValues);
-
-		const singleton =
-			collection.endsWith('_batch') === false &&
-			collection.endsWith('_items') === false &&
-			collection.endsWith('_item') === false &&
-			collection in this.schema.collections;
-
-		const single = collection.endsWith('_items') === false && collection.endsWith('_batch') === false;
-		const batchUpdate = action === 'update' && collection.endsWith('_batch');
-
-		if (collection.endsWith('_batch')) collection = collection.slice(0, -6);
-		if (collection.endsWith('_items')) collection = collection.slice(0, -6);
-		if (collection.endsWith('_item')) collection = collection.slice(0, -5);
-
-		if (singleton && action === 'update') {
-			return await this.upsertSingleton(collection, args['data'], query);
-		}
-
-		const service = getService(collection, {
-			knex: this.knex,
-			accountability: this.accountability,
-			schema: this.schema,
-		});
-
-		const hasQuery = (query.fields || []).length > 0;
-
-		try {
-			if (single) {
-				if (action === 'create') {
-					const key = await service.createOne(args['data']);
-					return hasQuery ? await service.readOne(key, query) : true;
-				}
-
-				if (action === 'update') {
-					const key = await service.updateOne(args['id'], args['data']);
-					return hasQuery ? await service.readOne(key, query) : true;
-				}
-
-				if (action === 'delete') {
-					await service.deleteOne(args['id']);
-					return { id: args['id'] };
-				}
-
-				return undefined;
-			} else {
-				if (action === 'create') {
-					const keys = await service.createMany(args['data']);
-					return hasQuery ? await service.readMany(keys, query) : true;
-				}
-
-				if (action === 'update') {
-					const keys: PrimaryKey[] = [];
-
-					if (batchUpdate) {
-						keys.push(...(await service.updateBatch(args['data'])));
-					} else {
-						keys.push(...(await service.updateMany(args['ids'], args['data'])));
-					}
-
-					return hasQuery ? await service.readMany(keys, query) : true;
-				}
-
-				if (action === 'delete') {
-					const keys = await service.deleteMany(args['ids']);
-					return { ids: keys };
-				}
-
-				return undefined;
-			}
-		} catch (err: any) {
-			return this.formatError(err);
-		}
+		return _getSchema(this, type);
```

Comment:
Not a fan of using kinda cryptic chars like `_` in function names, especially since  the underscore prefix in JS ecosystem usually means that something is not used.
Could we find a clearer name for this, maybe `getSchemaByType`, `getSchemaInternal`, ...? We might also just name it `getSchema` as well and use an import alias in this file to circumvent the naming conflict.